Userland/Libraries/LibGUI/VimEditingEngine.h

```diff
@@ -6,13 +6,28 @@
 
 #pragma once
 
+#include <AK/Function.h>
 #include <AK/Optional.h>
 #include <LibCore/Object.h>
 #include <LibGUI/EditingEngine.h>
 #include <LibGUI/TextRange.h>
 
 namespace GUI {
 
+class VimEditingEngine;
+
+enum VimMode {
+    Normal,
+    Insert,
+    Visual
+};
```

Comment:
Please avoid polluting the `GUI` namespace with enums. Use enum classes instead.
Note that this was fine when the enum was within `VimEditingEngine` because it was scoped to the class.